clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -378,15 +378,16 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
         memref.instr.type == TRACE_TYPE_PREFETCH_INSTR ||
         memref.instr.type == TRACE_TYPE_INSTR_NO_FETCH) {
         bool expect_encoding = TESTANY(OFFLINE_FILE_TYPE_ENCODINGS, shard->file_type_);
-        std::unique_ptr<instr_t> cur_instr_decoded = nullptr;
+        std::unique_ptr<instr_autoclean_t> cur_instr_decoded(
+            new instr_autoclean_t(GLOBAL_DCONTEXT, nullptr));
         if (expect_encoding) {
-            cur_instr_decoded.reset(new instr_t);
-            instr_init(GLOBAL_DCONTEXT, cur_instr_decoded.get());
-            app_pc next_pc = decode_from_copy(
-                GLOBAL_DCONTEXT, const_cast<app_pc>(memref.instr.encoding),
-                reinterpret_cast<app_pc>(memref.instr.addr), cur_instr_decoded.get());
+            cur_instr_decoded.get()->data = instr_create(GLOBAL_DCONTEXT);
+            app_pc next_pc = decode_from_copy(GLOBAL_DCONTEXT,
+                                              const_cast<app_pc>(memref.instr.encoding),
+                                              reinterpret_cast<app_pc>(memref.instr.addr),
+                                              cur_instr_decoded.get()->data);
```

Comment:
The .get() can be dropped if we're sure that the unique_ptr object is not nullptr (https://en.cppreference.com/w/cpp/memory/unique_ptr/operator*).

Reply:
Done.